File: week10/hard.py
def atbash_letter(letter: str) -> str:
    return chr(ord('a') + ord('z') - ord(letter))


def atbash_word(word: str) -> str:
    result = ""

    for letter in word:
        result += atbash_letter(letter)
    
    return result


def main():
    with open("input/hard.in", 'r') as reader, open("output/hard.out", 'w') as writer:
        is_next_line_encoded = False

        for line in reader:
            raw_word = line.strip()
            processed_word = atbash_word(raw_word) if is_next_line_encoded else raw_word
            writer.write(processed_word + '\n')

            is_next_line_encoded = not is_next_line_encoded



if __name__ == '__main__':
    main()

File: week10/Hard.java
import java.io.*;
import java.util.Scanner;

public class Hard {

    private static char atbash(char letter) {
        return (char) ('a' + 'z' - letter);
    }

    private static String atbash(String word) {
        char[] letters = word.toCharArray();

        for (int i = 0; i < letters.length; i++) {
            letters[i] = atbash(letters[i]);
        }

        return new String(letters);
    }

    public static void main(String[] args) {

        try (
            Scanner sc = new Scanner(new File("input/hard.in"));
            PrintWriter pw = new PrintWriter("output/hard.out")
        ) {

            boolean isNextLineEncoded = false;
            while (sc.hasNextLine()) {

                String rawWord = sc.nextLine();
                String processedWord = isNextLineEncoded ? atbash(rawWord) : rawWord;
                pw.println(processedWord);

                isNextLineEncoded = !isNextLineEncoded;
            }

        } catch (FileNotFoundException e) {
            System.err.println(e);
        }

    }

}
